File: Feb22/src/app/company.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Company } from './models/companies';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class CompanyService {

  
  httpUrl='http://localhost:3000/companies/';

  constructor(private httpClient:HttpClient) { }


  getAllCompanies():Observable<Company[]>{
    return this.httpClient.get<Company[]>(this.httpUrl);
  }
  
  saveCompany(company:Company):Observable<Company>{
    return this.httpClient.post<Company>(this.httpUrl,company);
  }
  
  deleteCompany(id:number):Observable<Company>{
    return this.httpClient.delete<Company>(this.httpUrl+ id);
  }
  updateCompanyInfo(company :Company) :Observable<Company>{
  
    return this.httpClient.put<Company>(this.httpUrl + company.id, company);
  }
  
  getCompanyById(id:number): Observable<Company>{
  
    return this.httpClient.get<Company>(this.httpUrl+id);
  
     
  }
}

File: march 3/company-service/src/main/java/com/cts/training/companyservice/CompanyService.java
package com.cts.training.companyservice;

import java.util.List;



public interface CompanyService {

	public List<CompanyDTO> getAllCompanies();
	public CompanyDTO getCompanyById(int id);
	public CompanyDTO saveCompany(CompanyDTO company);
	public void deleteCompany(int id);
	public CompanyDTO updateCompany(CompanyDTO company);
	public boolean activateCompany(String email);
	CompanyDTO insert(CompanyDTO companyDTO);
}
